### PR TITLE
extension for html preprocessors

### LINKED ISSUE #630: Provide EP for HTML preprocessors
Requires #628

Html renderer uses mechanism of preprocessors. Right now preprocessors are hardcoded. The goal of this task is to provide extension point in `dokka-base` for them and declare current proeprocessors as extensions.

### PR DESCRIPTION
**Pull request by sswistun-vl**
Extensions for html preprocessors. Fix #630

### CODE REVIEW

**Comment by Kordyjan** [comment]
Rebase yourself!

**Comment by Kordyjan** [comment]
Please rebase.

**Comment by sswistun-vl** [comment]
Rebased.

File: plugins/base/src/main/kotlin/DokkaBase.kt [hunk 1]
```diff
@@ -71,4 +73,28 @@ class DokkaBase : DokkaPlugin() {
...
+    val navigationPageInstaller by extending {
+        htmlPreprocessors with NavigationPageInstaller order { before(searchPageInstaller) }
+    }
+
+    val searchPageInstaller by extending {
+        htmlPreprocessors with SearchPageInstaller order { before(resourceInstaller) }
+    }
+
+    val resourceInstaller by extending {
+        htmlPreprocessors with ResourceInstaller order { before(styleAndScriptsAppender) }
```
Comment (kamilok1965) [inline plugins/base/src/main/kotlin/DokkaBase.kt]: Do we need the ordering here? Looks like those preprocessors are independent but I'm not sure

File: plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt [hunk 2]
```diff
@@ -8,6 +8,9 @@ import org.jetbrains.dokka.base.renderers.platforms
 import org.jetbrains.dokka.pages.*
 import org.jetbrains.dokka.transformers.pages.PageTransformer
 
+val defaultPreprocessors =
+    listOf(RootCreator, SearchPageInstaller, NavigationPageInstaller, ResourceInstaller, StyleAndScriptsAppender)
+
```
Comment (kamilok1965) [inline plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt]: This looks unused

Comment (sswistun-vl) [reply plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt]: Removed.

File: plugins/base/src/main/kotlin/DokkaBase.kt (hunk 1 above)
Comment (kamilok1965) [reply plugins/base/src/main/kotlin/DokkaBase.kt]: bump

File: plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt (hunk 2 above)
Comment (kamilok1965) [reply plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt]: Doesn't look like it

Comment (sswistun-vl) [reply plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt]: It should be now.

**Pull request merged**